### PR TITLE
Yyx/91 retain max words min freq

### PR DESCRIPTION
**Pull request by yixiangyin**
# Pull Request Details

## What issue does this PR address

- [retain max words min freq

- Link to associated issue: #91 

## Checklist

Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [ ] Screenshots included in linked issue
- [ ] Changes adhere to the style and coding guideline
- [ ] No confidential information
- [ ] No duplicated content
- [x] No lint check errors related to your changes (`make prep` or `flutter analyze lib`)
- [ ] Pre-exisiting lint errors noted: [HERE]
- [ ] Integration test `make qtest.tmp` screenshot included in issue
- [x] Tested on device:
  - [ ] Linux
  - [x] MacOS
  - [ ] Windows
- [x] Added two reviewers

## Finalising

Once PR discussion is complete and reviewers have approved:

- [ ] Merge dev into the branch
- [ ] Resolve any conflicts
- [ ] Add one line summary into CHANGELOG.md
- [ ] Bump appropriate version number in pubspec.yaml
- [ ] Push to git repository and review
- [ ] Merge PR into dev (gjwgit)

### CODE REVIEW

**Review by Lutra-Fs — APPROVED** [review]
LGTM.

File: lib/features/wordcloud/config.dart [hunk 1]
```diff
@@ -300,13 +304,27 @@ class _ConfigState extends ConsumerState<WordCloudConfig> {
...
+    if (txt != 'Inf') {
+      int? maxWord = int.tryParse(txt);
+      if (maxWord == null) {
+        return 'Inf';
+      }
+      return maxWord.toString();
+    }
+    return txt;
+  }
+
```
Comment (Lutra-Fs) [inline lib/features/wordcloud/config.dart]:
```suggestion
  String sanitiseMaxWord(String txt) {
    return txt == 'Inf' || int.tryParse(txt) != null ? txt : 'Inf';
  }
```

**Review by Lutra-Fs — CHANGES_REQUESTED** [review]

File: lib/features/wordcloud/config.dart [hunk 2]
```diff
@@ -300,13 +304,27 @@ class _ConfigState extends ConsumerState<WordCloudConfig> {
...
     debugPrint('max word text changed to ${maxWordTextController.text}');
-    ref.read(maxWordProvider.notifier).state = maxWordTextController.text;
+    ref.read(maxWordProvider.notifier).state =
+        sanitiseMaxWord(maxWordTextController.text);
   }
 
   void _updateMinFreqProvider() {
     debugPrint('min freq text changed to ${minFreqTextController.text}');
-    ref.read(minFreqProvider.notifier).state = minFreqTextController.text;
+    ref.read(minFreqProvider.notifier).state =
```
Comment (Lutra-Fs) [inline lib/features/wordcloud/config.dart:322]: If you reset it to 1, will the UI reflect that?

**Review by gjwgit — CHANGES_REQUESTED** [review]
Please review @Lutra-Fs suggestions.

Please remove debugPrint or use debugText.

File: lib/features/wordcloud/config.dart (hunk 1 above)
Comment (yixiangyin) [reply lib/features/wordcloud/config.dart]: Thanks for the suggestion.

File: lib/features/wordcloud/config.dart (hunk 2 above)
Comment (yixiangyin) [reply lib/features/wordcloud/config.dart:322]: yes

File: lib/r/start.dart [hunk 3]
```diff
@@ -100,7 +100,20 @@ void rStart(BuildContext context, WidgetRef ref) async {
   // screen. Let's see what it does on Windows.
 
   await Future(() async {
-    // Add the code to the SCRIPT tab.
+    // TODO yyx 20240926 merge conflict: should this part be included or no?
```
Comment (gjwgit) [inline lib/r/start.dart]: Keep here for now - working out what needs to be done.

Comment (gjwgit) [reply lib/r/start.dart]: I think we've moved it back into a branch. So remove it actually

**Review by gjwgit — APPROVED** [review]

**Pull request merged**